Link ncurses on ubuntu

I'll add an LDFLAG but is this something we should deal with?

https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/3022

https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/3022/steps/compile/logs/stdio
```
-- Found Readline: /usr/include  
-- Performing Test GNU_READLINE_FOUND
-- Performing Test GNU_READLINE_FOUND - Failed
-- Could not find GNU readline library so building without readline support
```

https://build.getmonero.org/builders/monero-static-ubuntu-amd64/builds/3022/steps/compile/logs/CMakeError
```
/usr/bin/c++    -DZMQ_STATIC -fno-strict-aliasing -maes -std=c++11 -D_GNU_SOURCE   -Wall -Wextra -Wpointer-arith -Wundef -Wvla -Wwrite-strings -Wno-error=extra -Wno-error=deprecated-declarations -Wno-unused-parameter -Wno-unused-variable -Wno-error=unused-variable -Wno-error=undef -Wno-error=uninitialized -Wlogical-op -Wno-error=maybe-uninitialized -Wno-error=cpp -Wno-reorder -Wno-missing-field-initializers -march=x86-64   -fno-strict-aliasing -DGNU_READLINE_FOUND    CMakeFiles/cmTC_e928c.dir/src.cxx.o  -o cmTC_e928c -rdynamic /usr/lib/x86_64-linux-gnu/libreadline.a 
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_get_screen_size':
(.text+0x201): undefined reference to `tgetnum'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_get_screen_size':
(.text+0x2cb): undefined reference to `tgetnum'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x647): undefined reference to `PC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x65a): undefined reference to `BC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x665): undefined reference to `UP'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x6f3): undefined reference to `tgetent'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x72f): undefined reference to `tgetstr'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x75b): undefined reference to `PC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x771): undefined reference to `BC'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x77f): undefined reference to `UP'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x78f): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x7d0): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x832): undefined reference to `tgetflag'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_init_terminal_io':
(.text+0x8fb): undefined reference to `tgetent'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_backspace':
(.text+0xad5): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `rl_ding':
(.text+0xb8f): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_enable_meta_key':
(.text+0xbf5): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_disable_meta_key':
(.text+0xc47): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o): In function `_rl_control_keypad':
(.text+0xc7b): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(terminal.o):(.text+0xcbe): more undefined references to `tputs' follow
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2418): undefined reference to `tgoto'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x242a): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2c26): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2e23): undefined reference to `tgoto'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2e33): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2ea1): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2ebd): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `update_line':
(.text+0x2eec): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o): In function `rl_redisplay':
(.text+0x4509): undefined reference to `tputs'
/usr/lib/x86_64-linux-gnu/libreadline.a(display.o):(.text+0x481c): more undefined references to `tputs' follow
collect2: error: ld returned 1 exit status
```

on OSX adding LDFLAGS="-ltermcap" works as shown in #2712

--- a/cmake/FindReadline.cmake
+++ b/cmake/FindReadline.cmake
@@ -36,14 +36,18 @@ find_library(Readline_LIBRARY
     NO_DEFAULT_PATH
 )
 
-if(Readline_INCLUDE_DIR AND Readline_LIBRARY AND Ncurses_LIBRARY)
+find_library(Termcap_LIBRARY
+  NAMES tinfo termcap ncursesw ncurses cursesw curses
+)
+
+if(Readline_INCLUDE_DIR AND Readline_LIBRARY)
   set(READLINE_FOUND TRUE)
-else(Readline_INCLUDE_DIR AND Readline_LIBRARY AND Ncurses_LIBRARY)
+else(Readline_INCLUDE_DIR AND Readline_LIBRARY)
   FIND_LIBRARY(Readline_LIBRARY NAMES readline PATHS Readline_ROOT_DIR)
   include(FindPackageHandleStandardArgs)
   FIND_PACKAGE_HANDLE_STANDARD_ARGS(Readline DEFAULT_MSG Readline_INCLUDE_DIR Readline_LIBRARY )
   MARK_AS_ADVANCED(Readline_INCLUDE_DIR Readline_LIBRARY)
-endif(Readline_INCLUDE_DIR AND Readline_LIBRARY AND Ncurses_LIBRARY)
+endif(Readline_INCLUDE_DIR AND Readline_LIBRARY)
 
 mark_as_advanced(
     Readline_ROOT_DIR
@@ -53,22 +57,25 @@ mark_as_advanced(
 
 set(CMAKE_REQUIRED_INCLUDES ${Readline_INCLUDE_DIR})
 set(CMAKE_REQUIRED_LIBRARIES ${Readline_LIBRARY})
-INCLUDE(CheckCXXSourceCompiles) 
-CHECK_CXX_SOURCE_COMPILES(
-"
-#include <stdio.h>
-#include <readline/readline.h>
-int
-main()
-{
-  char * s  = rl_copy_text(0, 0);
-}
-" GNU_READLINE_FOUND)
 
-if(NOT Readline_LIBRARY)
-  set(Readline_LIBRARY "")
-endif()
+include(CheckFunctionExists)
+check_function_exists(rl_copy_text HAVE_COPY_TEXT)
+check_function_exists(rl_filename_completion_function HAVE_COMPLETION_FUNCTION)
 
-if(Readline_LIBRARY AND OPENBSD)
-  list(APPEND EXTRA_LIBRARIES curses)
-endif()
+if(NOT HAVE_COMPLETION_FUNCTION)
+  unset(READLINE_FOUND)
+  set(CMAKE_REQUIRED_LIBRARIES ${Readline_LIBRARY} ${Termcap_LIBRARY})
+  check_function_exists(rl_copy_text HAVE_COPY_TEXT_TC)
+  check_function_exists(rl_filename_completion_function HAVE_COMPLETION_FUNCTION_TC)
+  set(HAVE_COMPLETION_FUNCTION ${HAVE_COMPLETION_FUNCTION_TC})
+  set(HAVE_COPY_TEXT ${HAVE_COPY_TEXT_TC})
+  if(HAVE_COMPLETION_FUNCTION)
+    set(Readline_LIBRARY ${Readline_LIBRARY} ${Termcap_LIBRARY})
+  endif(HAVE_COMPLETION_FUNCTION)
+endif(NOT HAVE_COMPLETION_FUNCTION)
+
+if(HAVE_COMPLETION_FUNCTION AND HAVE_COPY_TEXT)
+  set(GNU_READLINE_FOUND TRUE)
+  set(READLINE_FOUND TRUE)
+endif(HAVE_COMPLETION_FUNCTION AND HAVE_COPY_TEXT)
+

